### PR TITLE
inelastic background for isis instruments

### LINKED ISSUE #31327: Add empty background removal possibility to ISIS direct inelastic reduction
**Original reporter:** [STFC]/[<ross.stewart@stfc.ac.uk>]

New instruments have guides, which produce specific constant background during experiment.  The background is not affected by sample and can be measured on empty instrument

If this background is indeed measured, it can/should be removed from the results of inelastic data reduction

### PR DESCRIPTION
**Pull request by abuts**
**Description of work.**

There is bug or rather one missing flowpath in the previous PR [#31396](https://github.com/mantidproject/mantid/pull/31396). The sample workspace may indeed come normalized. There is an optimization within the diag I've forgotten about. 

**To test:**

Fixes #31327. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Inelastic/Direct/RunDescriptor.py [hunk 1]
```diff
@@ -1514,54 +1514,85 @@ def remove_empty_background(self,ebg_ws = None):
...
-        # Get dose for both workspaces
-        ebg_current = ebg_ws.run().getProperty('gd_prtn_chrg').value
-        ws_current = ws.run().getProperty('gd_prtn_chrg').value
-        # normalize by current and remove normalised background
-        ebg_ws  =  ebg_ws*(ws_current/ebg_current)
+        ws,run_normalised = self._check_normalised_and_normalise(ws,False)
+        ebg_ws,bg_normalised = self._check_normalised_and_normalise(ebg_ws,False)
+        if run_normalised or bg_normalised: # Both workspaces have to be normalized for correct operations
+            if not (run_normalised and bg_normalised):
+                ws,run_normalised = self._check_normalised_and_normalise(ws,True)
```
[the remark this answers is not in the document]
Comment (abuts) [reply scripts/Inelastic/Direct/RunDescriptor.py:1552]: the git view is confusing. This is how it should look: get current and prepare background workspace. 
```
        ws,run_normalised = self._check_normalised_and_normalise(ws,False)
        ebg_ws,bg_normalised = self._check_normalised_and_normalise(ebg_ws,False)
        if run_normalised or bg_normalised: # Both workspaces have to be normalized for correct operations
            if not (run_normalised and bg_normalised):
                ws,run_normalised = self._check_normalised_and_normalise(ws,True)
                ebg_ws,bg_normalised = self._check_normalised_and_normalise(ebg_ws,True)
        else:
            # Get dose for both workspaces. Normalisation will occur later
            ebg_current = ebg_ws.run().getProperty('gd_prtn_chrg').value
            ws_current = ws.run().getProperty('gd_prtn_chrg').value
            # normalize by current
            ebg_ws  =  ebg_ws*(ws_current/ebg_current)
        # End normalization check

        # remove normalised background
        Minus(ws,ebg_ws  ,OutputWorkspace = ws.name(),ClearRHSWorkspace=copy_created)
```

**Review by Harrietbrown — DISMISSED** [review]
Good Job

File: scripts/test/RunDescriptorTest.py [hunk 2]
```diff
@@ -646,11 +647,15 @@ def test_remove_background_matching_workspaces(self):
         self.assertTrue(difr.Result)
 
         # the subtaction occurs only once
-        PropertyManager.sample_run.remove_empty_background()
+        PropertyManager.sample_run.remove_empty_background(bgws)
         # operation above does nothing
         ws = PropertyManager.sample_run.get_workspace()
         difr = CompareWorkspaces(resWs,ws)
         self.assertTrue(difr.Result)
+        # ensure that bg Workspace remais unchanged
```
Comment (Harrietbrown) [inline scripts/test/RunDescriptorTest.py:655]:
```suggestion
        # ensure that bg Workspace remains unchanged
```

File: scripts/test/RunDescriptorTest.py [hunk 3]
```diff
@@ -743,9 +748,13 @@ def test_remove_background_matrix_minus_matrix(self):
         ref_ws = ref_ws - 0.1*ref_ws
         difr = CompareWorkspaces(ref_ws,resWs)
         self.assertTrue(difr.Result)
+        # bg Workspace remais unchanged
```
Comment (Harrietbrown) [inline scripts/test/RunDescriptorTest.py:751]:
```suggestion
        # bg Workspace remains unchanged
```

**Review by Harrietbrown — DISMISSED** [review]
Thanks for the additional tests.

**Review by Harrietbrown — APPROVED** [review]
Good Job

**Pull request merged**